toolkit/tools/scheduler/schedulerutils/preparerequest.go

```diff
@@ -47,10 +48,21 @@ func ConvertNodesToRequests(pkgGraph *pkggraph.PkgGraph, graphMutex *sync.RWMute
 	for _, nodes := range buildNodes {
 		const defaultNode = 0
 
+		// Check if any of the nodes in buildNodes is a delta node and mark it. We will use this to determine if the
+		// build is a delta build that might have pre-built .rpm files available.
+		hasADeltaNode := false
+		for _, node := range nodes {
+			if node.State == pkggraph.StateDelta {
+				hasADeltaNode = true
+				break
+			}
+		}
```

Comment:
Is it even possible for build nodes associated with the same SRPM to have different values of `node.State`? Wouldn't that be an indication of a bug in how we construct the graph?

Reply:
Hmm... good point. I'll change this to require all of the nodes match exactly.

Reply:
Tested this. We cannot guarantee that all nodes are delta. (for example, say we add a new subpackage and forget to bump the release. The subpackage .rpm will be missing from PMC and will NOT be a delta node, it will be a normal un-built node).